students/eric_grandeo/lesson09/mailroom_oo.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python3
+
+from collections import OrderedDict
+
+
+class Donor:
+    """
+    Class representing an individual donor
+
+    Args:
+        name (str):
+            Donor's name
+        donations(list):
+            Donation(s)
+
+    Attributes:
+        name (str):
+            Donor's name
+        donations (list):
+            List of donations, order is old to new
+        
+    Methods:
+        add_donation(amount):
+            Add donation to donation attribute list
+
+    Properties:
+        donor():
+            Return a dict of donor and donations
+        sum_donations():
+            Return total donations made by donor
+        num_donations():
+            Return the number of donations made by donor
+        avg_donation():
+            Return the average donation made by donor
+        thank_you_letter():
+            Generate text for email thanking donor for their last donation
+
+    """
+    def __init__(self, name, donations=[]):
+        self.name = name
+        self.donations = donations
+
+
+    @property
+    def donor(self):
```

Comment:
A common pattern is to have a method named `to_dict()` (makes it really clear what to expect